### PR TITLE
[MRG] add some docs on search, gather, and lca methods

### PR DESCRIPTION
**Pull request by ctb**
Update documentation significantly in preparation for 2.0 release -

* more complete command-line docs, including LCA subcommands;
* higher level info on classifying signatures;
* updated intro text highlighting functionality, limitations, and computational requirements;
* slightly cleaner command line foo.

Note, includes documentation for `lca gather`, introduced in #390. Incorporates PR #362 as well.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review & merge, @brooksph @taylorreiter @luizirber. 

While this is incomplete, it is a massive upgrade to the current documentation :). I'd like to suggest that review focus on accuracy, and that suggestions for the next round of improvement get added to a new issue. That way we can merge this sooner rather than later, while also keeping track of ideas for future improvements.

File: doc/classifying-signatures.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+Neither search option (similarity or containment) is effective when
+comparing or searching with metagenomes, which typically have a
+mixture of many different genomes.  While you might use containment to
+see if a query genome is present in one or more metagenomes, a common
+question to ask is the reverse: **what genomes are in my metagenome?**
+
+We have implemented two algorithms in sourmash to do this.
+
+One, approaches based on lowest common ancestor ("LCA"), uses
```
Comment (taylorreiter) [inline doc/classifying-signatures.md]: The language here is a little confusing, but not that critical to change

File: doc/classifying-signatures.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+Our preliminary benchmarking suggests that `gather` is the most accurate
+method available for doing strain-level resolution of genomes. More on that
+as we move forward!
+
+## To do taxonomy, or not to do taxonomy?
+
+By default, there is no structured taxonomic information available in
+sourmash signatures or SBT databases of signatures.  Generally what
+this means is that you will have to provide your own mapping from a
```
Comment (taylorreiter) [inline doc/classifying-signatures.md]: Does this refer to the output of gather? (Line 72-74)

File: doc/command-line.md [hunk 3]
```diff
@@ -109,16 +145,199 @@ Options:
...
+
+If the input signature was computed with `--track-abundance`, output
+will be abundance weighted (unless `--ignore-abundances` is
+specified).  `-o/--output` will create a CSV file containing the
+matches.
+
+`gather`, like `search`, will load all of provided signatures into
+memory.  You can use `sourmash index` to create a Sequence Bloom Tree
+(SBT) that can be quickly searched on disk; this is
+[the same format in which we provide GenBank and other databases](databases.html).
```
Comment (taylorreiter) [inline doc/command-line.md]: Line 205-207 is repeated above. I see the utility of the information being in both places, but is it repeated verbatim intentionally?

File: doc/command-line.md [hunk 4]
```diff
@@ -109,16 +145,199 @@ Options:
...
+genomes with no (or incomplete) taxonomic information.  Use `sourmash
+lca summarize` and `sourmash lca gather` to classify a metagenome
+using a collection of genomes with taxonomic information.
+
+## `sourmash lca` subcommands
+
+These commands use LCA databases (created with `index`, below, or
+prepared databases such as
+[genbank-k31.lca.json.gz, from the LCA tutorial](tutorials-lca.html).
+
```
Comment (taylorreiter) [inline doc/command-line.md]: I thought `index` was to create the SBT for gather?

File: doc/command-line.md [hunk 5]
```diff
@@ -109,16 +145,199 @@ Options:
...
+100.0%   792   Bacteria;Proteobacteria;Gammaproteobacteria;Alteromonadales;Shewanellaceae
+100.0%   792   Bacteria;Proteobacteria;Gammaproteobacteria;Alteromonadales
+100.0%   792   Bacteria;Proteobacteria;Gammaproteobacteria
+100.0%   792   Bacteria;Proteobacteria
+100.0%   792   Bacteria
+```
+
+### `sourmash lca gather`
+
+The `sourmash lca gather` command classifies finds all non-overlapping
```
Comment (taylorreiter) [inline doc/command-line.md]: classifies finds -- too many words I think

File: doc/classifying-signatures.md (hunk 1 above)
Comment (ctb) [reply doc/classifying-signatures.md]: Fixed! Thanks for noticing!

File: doc/classifying-signatures.md (hunk 2 above)
Comment (ctb) [reply doc/classifying-signatures.md]: No, it refers to all of the non-LCA stuff. This is just going to have to be confusing, given the addition of lca gather :).

File: doc/command-line.md (hunk 3 above)
Comment (ctb) [reply doc/command-line.md]: Yep!

File: doc/command-line.md (hunk 4 above)
Comment (ctb) [reply doc/command-line.md]: `lca index`. Fixed!

File: doc/command-line.md (hunk 5 above)
Comment (ctb) [reply doc/command-line.md]: Fixed!

**Comment by ctb** [comment]
@taylorreiter fixed in b10b490 - thx!

**Comment by ctb** [comment]
yay thanks! ...do you think we should wait for #390 to merge? I'm -0 on waiting.

**Comment by ctb** [comment]
(because this includes docs for `sourmash lca gather` which is not yet in master)

**Comment by ctb** [comment]
On Sun, Feb 18, 2018 at 10:20:29AM -0800, Taylor Reiter wrote:
> I don't have strong feelings either way, or a strong concept of the number of people we will confuse by merging now as opposed to waiting.
> 

yeah, me neither. Heck, I'll just merge it.

**Pull request merged**